### PR TITLE
♻️ Enhance regex validation director v2 -  Allow registries by IP and Port

### PR DESCRIPTION
**Pull request by mrnicegyu11**
## What do these changes do?
This makes a regex in the settings validation of director-v2 more permissive. The regex validates an env-var specifying the docker-registry which hosts the image for the dynamic sidecar.

Before the change, the following registry names were matched (and thus allowed):
- itisfoundation
- local

Now, additionally registries specified by IP& Port (e.g. `10.0.0.1:5000`) are allowed.

This change is necessary for deployments on multiple machines that run custom self-built development containers. For this multi-node setup, a self-hosted registry is necessary to deploy the images to the swarm. 

For regex validation, see: 

https://regex101.com/r/DoCQeV/1

### CODE REVIEW

File: services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py [hunk 1]
```diff
@@ -8,8 +8,11 @@
...
 DYNAMIC_PROXY_SERVICE_PREFIX = "dy-proxy"
 
-DYNAMIC_SIDECAR_DOCKER_IMAGE_RE = (
-    r"(^(local|itisfoundation)/)?(dynamic-sidecar):([\w]+)"
-)
+# This matches registries by:
+# - local
+# - itisfoundation
+# - 10.0.0.0:8473 (IP & Port)
+DYNAMIC_SIDECAR_DOCKER_IMAGE_RE = r"(^(local|itisfoundation|((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]):[0-9]+))/)?(dynamic-sidecar):([\w]+)"
```
Comment (KZzizzle) [inline services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py:15]: I'll let PC confirm this one - he dreams in regex

**Review by sanderegg — APPROVED** [review]

**Review by pcrespov — CHANGES_REQUESTED** [review]
- TIP: use composition with f-strings to make it more readable
- Make sure there are tests covering any new regex. It is specially critical for future changes (we can check together if you like)

File: services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py:15]: Sorry but this is way convoluted for what we need here.

Please use below expression:
```python
r"(^([a-zA-Z0-9:.-]+)/)?(dynamic-sidecar):([a-zA-Z0-9.]+)"
```

Add tests for dealing with the following cases:
- These should pass the regex
```
local/dynamic-sidecar:sadasd
itisfoundation/dynamic-sidecar:sadasd
10.10.10.10.no.ip:8080/dynamic-sidecar:10.0.1
10.10.10.10.no-ip:8080/dynamic-sidecar:sadasd
```
- this fails (note: `_` is not allowed in dns)
```
10.10.10.10.no_ip:8080/dynamic-sidecar:sadasd
10.10.10.10.no.ip:8080/dynamic-sidecar:the_tag
```

**Review by GitHK — CHANGES_REQUESTED** [review]
Please use my suggestion.
The previous regex was totally broken.
As PC requested, also add tests.

File: services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py (hunk 1 above)
Comment (mrnicegyu11) [reply services/director-v2/src/simcore_service_director_v2/models/schemas/constants.py]: Tests added, regex change, please re-review thx

**Review by GitHK — APPROVED** [review]
👍

File: services/director-v2/tests/unit/test_core_settings.py [hunk 2]
```diff
@@ -103,3 +109,24 @@ def test_dynamic_sidecar_settings(
...
+    image: str, project_env_devel_environment: Dict[str, Any]
+) -> None:
+    required_kwards = dict(
+        DYNAMIC_SIDECAR_IMAGE=image,
+        SIMCORE_SERVICES_NETWORK_NAME="test",
+        TRAEFIK_SIMCORE_ZONE="",
+        SWARM_STACK_NAME="",
+        DYNAMIC_SIDECAR_PROXY_SETTINGS=DynamicSidecarProxySettings(),
+    )
+    with pytest.raises(Exception) as exc_info:
```
Comment (pcrespov) [inline services/director-v2/tests/unit/test_core_settings.py:131]: ValidationError

**Review by pcrespov — APPROVED** [review]
thx for your contribution! :-)
ok with me but advice to implement changes we agreed .

**Pull request merged**